engine-tests/src/main/java/org/terasology/engine/core/PathManagerProvider.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.core;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+import static org.mockito.Mockito.mock;
+
+public class PathManagerProvider implements ParameterResolver {
+    private PathManagerProvider() {};
```

Comment:
wait, does this really work with the default constructor being private? that doesn't seem right

Reply:
don't you do this if you want to have a singleton object. is there a method to produce an instance :?.

Reply:
`PathManager::getInstance` is the method that creates the instance in a singleton fashion I guess.
@pollend why would one do `private PathManagerProvider() {};` if one would want to have a singleton object? (this is just me not knowing much about java I guess)

Reply:
:books: _hands over Gang of Four book to Nira ..._

You want to ensure that there is only one instance of this class, so you need to restrict who can create instances. The typical (Java) pattern here is to make the (or rather all) constructor private to that class, and than offer a static method to get the single instance `getInstance`. This static method is allowed to call the private constructor, so you're good to go.

(there are numerous SO threads on the topic on whether this is thread safe, etc)

Reply:
To follow up on the code at hand: my guess would be on JUnit/Mockito magic around reflection and making impossible things possible for testing :mage_man: ...